compatibility/cck_spec.ts

```diff
@@ -0,0 +1,97 @@
+import { describe, it } from 'mocha'
+import { config, expect, use } from 'chai'
+import chaiExclude from 'chai-exclude'
+import globby from 'globby'
+import fs from 'fs'
+import ndjsonParse from 'ndjson-parse'
+import path from 'path'
+import { PassThrough } from 'stream'
+import { Cli } from '../lib'
+import toString from 'stream-to-string'
+import { doesHaveValue, doesNotHaveValue } from '../src/value_checker'
+import { normalizeMessageOutput } from '../features/support/formatter_output_helpers'
+
+const PROJECT_PATH = path.join(__dirname, '..')
+const CCK_FEATURES_PATH = 'node_modules/@cucumber/compatibility-kit/features'
+const CCK_IMPLEMENTATIONS_PATH = 'compatibility/features'
+
+config.truncateThreshold = 100
+use(chaiExclude)
+
+describe('Cucumber Compatibility Kit', () => {
+  globby.sync([`${CCK_FEATURES_PATH}/**/*.ndjson`]).forEach((fixturePath) => {
+    const suiteName = /^.+\/(.+)\.ndjson$/.exec(fixturePath)[1]
+    it(`passes the cck suite for '${suiteName}'`, async () => {
+      const args = [
+        'node',
+        path.join(PROJECT_PATH, 'bin', 'cucumber-js'),
+      ].concat([
+        `${CCK_FEATURES_PATH}/${suiteName}/${suiteName}.feature`,
+        '--require',
+        `${CCK_IMPLEMENTATIONS_PATH}/${suiteName}/${suiteName}.ts`,
+        '--profile',
+        'cck',
+      ])
+      const stdout = new PassThrough()
+      try {
+        await new Cli({
+          argv: args,
+          cwd: PROJECT_PATH,
+          stdout,
+        }).run()
+      } catch (ignored) {
+        console.error(ignored)
+      }
+      stdout.end()
+
+      const rawOutput = await toString(stdout)
+      const actualMessages = normalize(ndjsonParse(rawOutput))
+      const expectedMessages = ndjsonParse(
+        fs.readFileSync(fixturePath, { encoding: 'utf-8' })
+      )
+      expect(actualMessages)
+        .excludingEvery([
+          'meta',
+          // sources
+          'uri',
+          'line',
+          // ids
+          'astNodeId',
+          'astNodeIds',
+          'hookId',
+          'id',
+          'pickleId',
+          'pickleStepId',
+          'stepDefinitionIds',
+          'testCaseId',
+          'testCaseStartedId',
+          'testStepId',
+          // time
+          'nanos',
+          'seconds',
+          // errors
+          'message',
```

Comment:
Are we expecting all of these to be excluded long term? I would hope with predictable ids that big block could be fixed.  Do you think its worth having an issue to try to drive this list down.

Reply:
I agree - when I was working against a slightly earlier CDK version I think it was only the `testCase` ordering that made the ids not line up.

But cck has [changed](https://github.com/cucumber/cucumber/commit/e254f6a9ba2c41e52485a76c1f78221cef63879a#diff-102d4a7789d087b284dbfed6836597fd) from using predictable ids to uuids so not sure where that leaves us.

Reply:
The predictable ids and timestamps are a dead end. The cck no longer uses them.

It's a dead end because various Cucumber implementations emit messages in slightly different order than what `fake-cucumber` (the engine behind the cck) emits. It would be too much work to make them all emit in the same order.

So what we opted for instead is a validation mechanism where certain values (ids, timestamps) are not compared. We pair this with smoke tests that send the generated messages through 2 of the most advanced message consumers we have - cucumber-html and json-formatter. If these don't crash and a cursory eye-balling looks OK, we're confident enough that the emitted messages are ok.